### PR TITLE
Dl-grid-refactor

### PR DESCRIPTION
**Pull request by prodan-eric**

### CODE REVIEW

**Comment by fadiDL** [comment]
@RotemDl please give insight on if this is enough

**Comment by prodan-eric** [comment]
update: After the meeting there's some more things to be fixed:
1.  swapping widgets should also affect the grid layout
2. the order of the widgets should somehow be emitted in order to be saved

**Comment by VictorOpenTech** [comment]
Issues related to > https://opentechdev.atlassian.net/browse/DLCL-879

**Review by fadiDL — CHANGES_REQUESTED** [review]
@prodan-eric two main issues noticed in the last version here
1. it was very difficult to swap trigger a layout change when moving between one row to another ( add skeleton animation and make hover faster to show changes )
2. dont allow swappign between widgets
3. add limitation per row for widgets ( no more than 3 for example ) 
4. add think of a demo for dl-grid that contains images from the internet to have something like google images search. ( dynamically calculate itself ) ( this should also add a new prop called fixed where it manually handles all changes and does not allow dragging  of inner items ) 

Great job on the layout change the v-model works great

**Comment by prodan-eric** [comment]
> @prodan-eric two main issues noticed in the last version here @vadim-popa-opentech
> 
> 1. it was very difficult to swap trigger a layout change when moving between one row to another ( add skeleton animation and make hover faster to show changes )
> 2. dont allow swappign between widgets
> 3. add limitation per row for widgets ( no more than 3 for example ) ( should be a prop of max-per-row )
>    4.think of a demo for dl-grid that contains images from the internet to have something like google images search. ( dynamically calculate itself ) ( this should also add a new prop called fixed where it manually handles all changes and does not allow dragging  of inner items )
> 
> Great job on the layout change the v-model works great

There is a prop for limitation with a default of 3. I just have a few question: What do you mean by skeleton animation ?

**Review by fadiDL — CHANGES_REQUESTED** [review]
Please fix conflicts with main branch and make the branch ready for merging

File: src/components/basic/DlWidget/DlWidget.vue [hunk 1]
```diff
@@ -7,24 +7,22 @@
...
                     :style="`visibility: ${
-                        visibleDragIcon ? 'visible' : 'hidden'
+                        visibleDragIcon && !isDragging ? 'visible' : 'hidden'
                     }`"
                     class="dl-widget__header--drag-icon"
                     icon="icon-dl-drag"
                     color="dl-color-medium"
                     size="15px"
-                    @mousedown="startDragging"
+                    @mousedown.native="startDragging"
```
Comment (fadiDL) [inline src/components/basic/DlWidget/DlWidget.vue:25]: @prodan-eric why do we need .native here? we've used `@mousedown` on dlchip before with no issue and we didnt need to use native. if not having native causes issues please add emitting of the mouse events from the root container in the dl-chip itself and try to not use the deprecated `.native` modifier.

File: tests/DlSmartSearch/DlSmartSearchInput.spec.ts [hunk 2]
```diff
@@ -31,10 +31,6 @@ describe('DlSmartSearchInput', () => {
             status: { type: 'info', message: 'info label' }
         })
 
-        wrapper.vm.focused = true
```
Comment (fadiDL) [inline tests/DlSmartSearch/DlSmartSearchInput.spec.ts:34]: Why is this removed btw ? @prodan-eric

**Review by fadiDL — CHANGES_REQUESTED** [review]

File: tests/DlSmartSearch/DlSmartSearchInput.spec.ts (hunk 2 above)
Comment (prodan-eric) [reply tests/DlSmartSearch/DlSmartSearchInput.spec.ts:34]: Not sure, I added it back

File: src/components/basic/DlWidget/DlWidget.vue (hunk 1 above)
Comment (prodan-eric) [reply src/components/basic/DlWidget/DlWidget.vue:25]: I removed it

**Review by fadiDL — APPROVED** [review]
